File: cf/2.go
package main

import (
	"fmt"
	"math"
)

type Point struct {
	X, Y int
}

func euclideanDistance(p1, p2 Point) float64 {
	return math.Sqrt(math.Pow(float64(p1.X-p2.X), 2) + math.Pow(float64(p1.Y-p2.Y), 2))
}

func farthestPoints(points []Point) map[int]int {
	farthestPoints := make(map[int]int)
	for i, p1 := range points {
		maxDistance := 0.0
		var farthestPointID int
		for j, p2 := range points {
			if i != j { 
				distance := euclideanDistance(p1, p2)
				if distance > maxDistance {
					maxDistance = distance
					farthestPointID = j + 1
				}
			}
		}
		farthestPoints[i+1] = farthestPointID
	}
	return farthestPoints
}

func main() {
	var N int
	fmt.Scan(&N)

	points := make([]Point, N)
	for i := 0; i < N; i++ {
		var x, y int
		fmt.Scan(&x, &y)
		points[i] = Point{X: x, Y: y}
	}

	farthestPointsIDs := farthestPoints(points)
	for _, farthestPointID := range farthestPointsIDs {
		fmt.Println(farthestPointID)
	}
}

File: cf/2.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

struct Point
{
    int x, y;
};

double euclideanDistance(const Point &p1, const Point &p2)
{
    return sqrt(pow(p1.x - p2.x, 2) + pow(p1.y - p2.y, 2));
}

pair<int, double> farPoint(const vector<Point> &p)
{
    int maxIndex = 0;
    double maxDistance = 0.0;
    for (int i = 0; i < p.size(); i++)
    {
        for (int j = i + 1; j < p.size(); j++)
        {
            double d = euclideanDistance(p[i], p[j]);
            if (d > maxDistance)
            {
                maxDistance = d;
                maxIndex = j;
            }
        }
    }
    return make_pair(maxIndex, maxDistance);
}

int main()
{
    int n;
    cin >> n;
    vector<Point> p(n);
    for (int i = 0; i < n; i++)
    {
        cin >> p[i].x >> p[i].y;
    }

    pair<int, double> result = farPoint(p);
    cout << result.first << endl;

    return 0;
}
